app/knight.py

```diff
@@ -0,0 +1,40 @@
+class Knight:
+    protection = 0

```

Comment:
There is no need to have `protection` here as a class attribute because it rather relates to an instance, place it in the `__init__`. Besides that, you are calling `protection` via `self` in code, which is why `__init__` is a better place to have this attribute.